python/ray/dag/dag_node.py

```diff
@@ -84,14 +91,27 @@ def _collect_upstream_nodes(self) -> List["DAGNode"]:
         them up instead of reference counting. We should consider using weak references
         to avoid circular references.
         """
+        upstream_nodes: List["DAGNode"] = []
+
+        assert hasattr(self._bound_args, "__iter__")
+        for arg in self._bound_args:
```

Comment:
Can you add inline comments regarding the purpose of doing this

Reply:
updated https://github.com/ray-project/ray/pull/48302/commits/2b9f24e908787ad83914bc1c383e19e10bf4cb75